### PR TITLE
Updated error message when saving WebP with invalid width or height

### LINKED ISSUE #8321: `ValueError: encoding error 5` during WEBP encoding should be replaced with something more informative
### What did you do?

I tried to save an 4x upscaled image with WEBP format exceeding the WEBP image size limit

### What did you expect to happen?

~~I expected the image to be saved~~
The error message should be more informative, it should say something like `The image is too big for the WEBP format`

### What actually happened?

Program crashed with:
```pytb
...
File "C:\Program Files\Python312\Lib\site-packages\PIL\Image.py", line 2439, in save
    save_handler(self, fp, filename)
  File "C:\Program Files\Python312\Lib\site-packages\PIL\WebPImagePlugin.py", line 340, in _save
    data = _webp.WebPEncode(
           ^^^^^^^^^^^^^^^^^
ValueError: encoding error 5
```

### What are your OS, Python and Pillow versions?

* OS: Windows 11 Pro 23H2
* Python: Python 3.12.1
* Pillow: 10.2.0 & 10.4.0

```text
--------------------------------------------------------------------
Pillow 10.2.0
Python 3.12.1 (tags/v3.12.1:2305ca5, Dec  7 2023, 22:03:25) [MSC v.1937 64 bit (AMD64)]
--------------------------------------------------------------------
Python modules loaded from C:\Program Files\Python312\Lib\site-packages\PIL
Binary modules loaded from C:\Program Files\Python312\Lib\site-packages\PIL
--------------------------------------------------------------------
--- PIL CORE support ok, compiled for 10.2.0
--- TKINTER support ok, loaded 8.6
--- FREETYPE2 support ok, loaded 2.13.2
--- LITTLECMS2 support ok, loaded 2.16
--- WEBP support ok, loaded 1.3.2
--- WEBP Transparency support ok
--- WEBPMUX support ok
--- WEBP Animation support ok
--- JPEG support ok, compiled for libjpeg-turbo 3.0.1
--- OPENJPEG (JPEG2000) support ok, loaded 2.5.0
--- ZLIB (PNG/ZIP) support ok, loaded 1.3
--- LIBTIFF support ok, loaded 4.6.0
*** RAQM (Bidirectional Text) support not installed
*** LIBIMAGEQUANT (Quantization method) support not installed
*** XCB (X protocol) support not installed
--------------------------------------------------------------------
BLP
Extensions: .blp
Features: open, save, encode
--------------------------------------------------------------------
BMP image/bmp
Extensions: .bmp
Features: open, save
--------------------------------------------------------------------
BUFR
Extensions: .bufr
Features: open, save
--------------------------------------------------------------------
CUR
Extensions: .cur
Features: open
--------------------------------------------------------------------
DCX
Extensions: .dcx
Features: open
--------------------------------------------------------------------
DDS
Extensions: .dds
Features: open, save
--------------------------------------------------------------------
DIB image/bmp
Extensions: .dib
Features: open, save
--------------------------------------------------------------------
EPS application/postscript
Extensions: .eps, .ps
Features: open, save
--------------------------------------------------------------------
FITS
Extensions: .fit, .fits
Features: open
--------------------------------------------------------------------
FLI
Extensions: .flc, .fli
Features: open
--------------------------------------------------------------------
FTEX
Extensions: .ftc, .ftu
Features: open
--------------------------------------------------------------------
GBR
Extensions: .gbr
Features: open
--------------------------------------------------------------------
GIF image/gif
Extensions: .gif
Features: open, save, save_all
--------------------------------------------------------------------
GRIB
Extensions: .grib
Features: open, save
--------------------------------------------------------------------
HDF5
Extensions: .h5, .hdf
Features: open, save
--------------------------------------------------------------------
ICNS image/icns
Extensions: .icns
Features: open, save
--------------------------------------------------------------------
ICO image/x-icon
Extensions: .ico
Features: open, save
--------------------------------------------------------------------
IM
Extensions: .im
Features: open, save
--------------------------------------------------------------------
IMT
Features: open
--------------------------------------------------------------------
IPTC
Extensions: .iim
Features: open
--------------------------------------------------------------------
JPEG image/jpeg
Extensions: .jfif, .jpe, .jpeg, .jpg
Features: open, save
--------------------------------------------------------------------
JPEG2000 image/jp2
Extensions: .j2c, .j2k, .jp2, .jpc, .jpf, .jpx
Features: open, save
--------------------------------------------------------------------
MCIDAS
Features: open
--------------------------------------------------------------------
MPEG video/mpeg
Extensions: .mpeg, .mpg
Features: open
--------------------------------------------------------------------
MSP
Extensions: .msp
Features: open, save, decode
--------------------------------------------------------------------
PCD
Extensions: .pcd
Features: open
--------------------------------------------------------------------
PCX image/x-pcx
Extensions: .pcx
Features: open, save
--------------------------------------------------------------------
PIXAR
Extensions: .pxr
Features: open
--------------------------------------------------------------------
PNG image/png
Extensions: .apng, .png
Features: open, save, save_all
--------------------------------------------------------------------
PPM image/x-portable-anymap
Extensions: .pbm, .pgm, .pnm, .ppm
Features: open, save
--------------------------------------------------------------------
PSD image/vnd.adobe.photoshop
Extensions: .psd
Features: open
--------------------------------------------------------------------
QOI
Extensions: .qoi
Features: open
--------------------------------------------------------------------
SGI image/sgi
Extensions: .bw, .rgb, .rgba, .sgi
Features: open, save
--------------------------------------------------------------------
SPIDER
Features: open, save
--------------------------------------------------------------------
SUN
Extensions: .ras
Features: open
--------------------------------------------------------------------
TGA image/x-tga
Extensions: .icb, .tga, .vda, .vst
Features: open, save
--------------------------------------------------------------------
TIFF image/tiff
Extensions: .tif, .tiff
Features: open, save, save_all
--------------------------------------------------------------------
WEBP image/webp
Extensions: .webp
Features: open, save, save_all
--------------------------------------------------------------------
WMF
Extensions: .emf, .wmf
Features: open, save
--------------------------------------------------------------------
XBM image/xbm
Extensions: .xbm
Features: open, save
--------------------------------------------------------------------
XPM image/xpm
Extensions: .xpm
Features: open
--------------------------------------------------------------------
XVTHUMB
Features: open
--------------------------------------------------------------------
```

```python
# coding=utf-8
import PIL.Image

if __name__ == "__main__":
    image = PIL.Image.open(r"..\..\input\webp_test.png")

    w, h = image.size
    scaled_image = image.resize((w * 4, h * 4), PIL.Image.Resampling.LANCZOS)

    optional_args = {
        'lossless': True,
        'method': 6,
        'optimize': True,
        'format': 'WEBP'
    }
    scaled_image.save(r"..\..\output\webp_test.webp", **optional_args)
```
webp_test

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #8321

There is a limit on the size of WebP images.

https://developers.google.com/speed/webp/faq
> The maximum pixel dimensions of a WebP image is 16383 x 16383.

When that limit is hit, this PR changes the error message from
"encoding error 5"
to
"encoding error 5: Image size exceeds WebP limit"

### CODE REVIEW

**Review by hugovk — APPROVED** [review]
Worth mentioning what the limit is? Or leave it out in case it might change?

**Comment by radarhere** [comment]
I've updated the message to "encoding error 5: Image size exceeds WebP limit of 16383", using `WEBP_MAX_DIMENSION` so that if WebP does change the value, the message will also update.

File: src/_webp.c [hunk 1]
```diff
@@ -672,7 +672,14 @@ WebPEncode_wrapper(PyObject *self, PyObject *args) {
 
     WebPPictureFree(&pic);
     if (!ok) {
-        PyErr_Format(PyExc_ValueError, "encoding error %d", (&pic)->error_code);
+        int error_code = (&pic)->error_code;
+        char message[50] = "";
+        if (error_code == VP8_ENC_ERROR_BAD_DIMENSION) {
+            sprintf(
+                message, ": Image size exceeds WebP limit of %d", WEBP_MAX_DIMENSION
```
Comment (hugovk) [inline src/_webp.c:679]: Include units?
```suggestion
                message, ": Image size exceeds WebP limit of %d pixels", WEBP_MAX_DIMENSION
```

File: Tests/test_file_webp.py [hunk 2]
```diff
@@ -157,6 +157,16 @@ def test_write_encoding_error_message(self, tmp_path: Path) -> None:
...
         assert str(e.value) == "encoding error 6"
 
+    @pytest.mark.skipif(sys.maxsize <= 2**32, reason="Requires 64-bit system")
+    def test_write_encoding_error_bad_dimension(self, tmp_path: Path) -> None:
+        temp_file = str(tmp_path / "temp.webp")
+        im = Image.new("L", (16384, 16384))
+        with pytest.raises(ValueError) as e:
+            im.save(temp_file)
+        assert (
+            str(e.value) == "encoding error 5: Image size exceeds WebP limit of 16383"
```
Comment (hugovk) [inline Tests/test_file_webp.py:167]:
```suggestion
            str(e.value) == "encoding error 5: Image size exceeds WebP limit of 16383 pixels"
```

**Review by hugovk — APPROVED** [review]

**Pull request merged**